c2rust-analyze/src/borrowck/type_check.rs

```diff
@@ -161,10 +164,63 @@ impl<'tcx> TypeChecker<'tcx, '_> {
         lty
     }
 
-    pub fn visit_operand(&self, op: &Operand<'tcx>) -> LTy<'tcx> {
+    pub fn visit_operand(&mut self, op: &Operand<'tcx>) -> LTy<'tcx> {
         match *op {
             Operand::Copy(pl) | Operand::Move(pl) => self.visit_place(pl),
             Operand::Constant(ref c) => {
+                if c.ty().is_any_ptr() {
+                    if let Some(alloc_id) = const_alloc_id(c) {
+                        if let Some(did) = find_static_for_alloc(&self.acx.tcx(), alloc_id) {
+                            let lty = assign_origins(
+                                self.ltcx,
+                                self.hypothesis,
+                                self.facts,
+                                self.maps,
+                                &self.acx.gacx.adt_metadata,
+                                &self.acx.gacx.static_tys[&did],
+                            );
+
+                            for l in lty.iter() {
+                                let static_origin = self.static_origin;
+                                let mut add_subset_base = |pl: Origin, rv: Origin| {
+                                    let point = self.current_point(SubPoint::Mid);
+                                    self.facts.subset_base.push((rv, pl, point));
+                                };
+                                if let Some(origin) = l.label.origin {
+                                    // constrain this origin to be 'static
+                                    eprintln!("constraining origin {origin:?} to 'static lifetime");
+                                    add_subset_base(static_origin, origin);
+                                    add_subset_base(origin, static_origin)
+                                }
+                                for (op, origin) in l.label.origin_params {
+                                    // constrain this origin to be 'static
+                                    eprintln!("constraining origin {op:?} ({origin:?}) to 'static lifetime");
+                                    add_subset_base(static_origin, *origin);
+                                    add_subset_base(*origin, static_origin);
+                                }
+                            }
+
+                            eprintln!("NEW STATIC LTY: {:?}", c.ty().kind());
+                            let pointer_id = self.acx.type_of(op).label;
+                            let perm = self.hypothesis[pointer_id];
+                            let args = self.ltcx.mk_slice(&[lty]);
+                            assert!(matches!(
+                                *c.ty().kind(),
+                                TyKind::Ref(..) | TyKind::RawPtr(..)
+                            ));
+
+                            // Polonius does not appear to issue loans for statics,
+                            // but assigns an origin
```

Comment:
```suggestion
                            // Polonius does not appear to issue loans for the outermost ref/ptr in
                            // an address-of-static constant, but it does assign an origin
```
Small edit to clarify why the origin here is handled differently from the origins produced by `assign_origins` above.